### PR TITLE
Custom op export test template

### PR DESCRIPTION
**Pull request by neginraoof**
Fixes for pipeline template to test pytorch export of custom ops

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/linux-ort-pytorch-ci-pipeline.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
     submodules: recursive
 
   - script: >
-      python -m ensurepip
+      sudo apt-get install -y python3-pip
```
Comment (thiagocrepaldi) [inline tools/ci_build/github/azure-pipelines/linux-ort-pytorch-ci-pipeline.yml:15]: Most pipelines seems to be creating docker images to run build / tests. By installing packages directly on the server, we can break the whole environment. Each host can be ubuntu 16.04, ubuntu 18.04, centos, etc, so package version/availability could differ, which can cause hard to repro issues.

Is it possible to use `tools/ci_build/github/linux/run_dockerbuild.sh` to create a docker image and `docker run` to execute the tests as other pipelines? In these images you can install everything you need without compromising the host server

https://github.com/microsoft/onnxruntime/blob/master/tools/ci_build/github/azure-pipelines/orttraining-linux-gpu-e2e-test-ci-pipeline.yml

and

https://github.com/microsoft/onnxruntime/blob/master/tools/ci_build/github/azure-pipelines/orttraining-linux-gpu-frontend-test-ci-pipeline.yml

might serve as examples

File: tools/ci_build/github/linux/test_custom_ops_pytorch_export.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export PYTHONPATH='/home/neraoof/onnxruntime/tools'
+
+
+pytest /home/neraoof/onnxruntime/tools/test/test_custom_ops_pytorch_exporter.py || exit 1
```
Comment (thiagocrepaldi) [inline tools/ci_build/github/linux/test_custom_ops_pytorch_export.sh:4]: hard coded path for your home directory "/home/neraoof"

Comment (neginraoof) [reply tools/ci_build/github/linux/test_custom_ops_pytorch_export.sh]: Thanks. I'm doing some tests on pipeline to make sure we handle failures in script and tests correctly. I'm going to mark this as WIP for now until I'm done with trial and error.

File: tools/ci_build/github/azure-pipelines/linux-ort-pytorch-ci-pipeline.yml (hunk 1 above)
Comment (neginraoof) [reply tools/ci_build/github/azure-pipelines/linux-ort-pytorch-ci-pipeline.yml]: Thanks for this suggestion, this is very helpful. I'm not familiar with setting up docker pipelines. I'm doing some trial and error editing the template and finding out how it works.

Comment (neginraoof) [reply tools/ci_build/github/azure-pipelines/linux-ort-pytorch-ci-pipeline.yml]: @thiagocrepaldi 
I've added the step to run docker build, but when running the test script I'm seeing this error:
ModuleNotFoundError: No module named 'onnxruntime'

Do you know if another step is needed to install the wheel?

File: tools/ci_build/github/linux/test_custom_ops_pytorch_export.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+pip3 install --user numpy==1.19.0 torch pytest
+
+export PYTHONPATH="/onnxruntime_src/tools"
```
Comment (thiagocrepaldi) [inline tools/ci_build/github/linux/test_custom_ops_pytorch_export.sh:4]: not sure whether this is your issue or not, but usually we append content to PYTHONPATH as opposed to override it. Try

`export PYTHONPATH="/onnxruntime_src/tools:${PYTHONPATH}"`

File: tools/ci_build/github/azure-pipelines/linux-ort-pytorch-ci-pipeline.yml [hunk 4]
```diff
@@ -1,16 +1,29 @@
...
+
   steps:
   - checkout: self
     clean: true
     submodules: recursive
 
   - script: >
-      PYTHONPATH='$(Build.SourcesDirectory)/tools' pytest -v python tools/python/test_custom_ops_pytorch_exporter.py
-    displayName: 'Test Custom Ops PyTorch Export'
+      tools/ci_build/github/linux/run_dockerbuild.sh
```
Comment (thiagocrepaldi) [inline tools/ci_build/github/azure-pipelines/linux-ort-pytorch-ci-pipeline.yml:14]: I dont remember what is inside `tools/ci_build/github/linux/run_dockerbuild.sh`, but I think it only builds ort and creates the wheel package. You may have to do a `pip install $(Build.BinariesDirectory)/dist/*.whl` or something like that

Try running this script on your machine and then do a local `docker run` to test whether ort is installed

File: tools/ci_build/github/azure-pipelines/linux-ort-pytorch-ci-pipeline.yml (hunk 1 above)
Comment (thiagocrepaldi) [reply tools/ci_build/github/azure-pipelines/linux-ort-pytorch-ci-pipeline.yml]: Hi @neginraoof I am not sur ewhether we spoke about this, but you are probably missing a pip install <ort>/build/Linux/<build_type>/dist/onnxruntime*.whl to install ort on your docker image

another possibility is using `export PYTHONPATH:<ort>/build/Linux/<build_type>:$PYTHONPATH` before running your script

**Comment by neginraoof** [comment]
@thiagocrepaldi Thanks a lot. I finally got the script working. This is a sample CI: https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=128324&view=results

**Pull request merged**